docs: start feature parity section (#110)
Coly010/ng-rspack-build: README.md

| ---------------------------------------------------------------------------------------------------------------- | --------------------------------------------------------------------------------------------------- | --------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------ |
| [@ng-rspack/build](https://github.com/Coly010/ng-rspack-build/tree/main/packages/build)                          | [@angular/build](https://github.com/angular/angular/tree/main/packages/build)                       | [@angular-devkit/build-angular](https://github.com/angular/angular-cli/tree/main/packages/angular_devkit) | Core build system for ng-rspack, replacing Angular CLI's Webpack-based builder.                  |

> | [@ng-rspack/compiler](https://github.com/Coly010/ng-rspack-build/tree/main/packages/compiler)                    | [@angular/build](https://github.com/angular/angular/tree/main/packages/build)                       | [@angular/compiler](https://github.com/angular/angular/tree/main/packages/compiler)                       | Compiler for Angular applications using Rspack, leveraging abstractions from `@angular/build`.   |

Reviewer: ```suggestion
| [@ng-rspack/compiler](https://github.com/Coly010/ng-rspack-build/tree/main/packages/compiler)                    | [@angular/build](https://github.com/angular/angular/tree/main/packages/build)                       | [@angular/build](https://github.com/angular/angular/tree/main/packages/compiler)                       | Compiler for Angular applications using Rspack, leveraging abstractions from `@angular/build`.   |
```

> | [@ng-rspack/nx](https://github.com/Coly010/ng-rspack-build/tree/main/packages/nx-plugin)                         | n/a ([@nrwl/angular](https://github.com/nrwl/nx/tree/master/packages/angular) as already 3rd party) | n/a ([@nrwl/angular](https://github.com/nrwl/nx/tree/master/packages/angular) as already 3rd party)       | Provides Nx integration for RsPack and RsBuild in Angular, like `@nrwl/angular` for Angular CLI. |

Reviewer 2: ```suggestion
| [@ng-rspack/nx](https://github.com/Coly010/ng-rspack-build/tree/main/packages/nx-plugin)                         | n/a ([@nx/angular](https://github.com/nrwl/nx/tree/master/packages/angular) as already 3rd party) | n/a ([@nx/angular](https://github.com/nrwl/nx/tree/master/packages/angular) as already 3rd party)       | Provides Nx integration for RsPack and RsBuild in Angular, like `@nx/angular` for Angular CLI. |
```

| [@ng-rsbuild/plugin-angular](https://github.com/Coly010/ng-rspack-build/tree/main/packages/build-plugin-angular) | [@angular/build](https://github.com/angular/angular/tree/main/packages/build) (Builder API)         | [@angular-devkit/build-angular](https://github.com/angular/angular-cli/tree/main/packages/angular_devkit) | Rsbuild plugin for Angular projects, similar to Angular CLI's Webpack-based builder API.         |

> | [@ng-rsbuild/plugin-nx](https://github.com/Coly010/ng-rspack-build/tree/main/packages/build-plugin-nx)           | [@nrwl/angular](https://github.com/nrwl/nx/tree/master/packages/angular)                            | [@nrwl/angular](https://github.com/nrwl/nx/tree/master/packages/angular)                                  | Provides Nx integration for RsPack and RsBuild in Angular, like `@nrwl/angular` for Angular CLI. |

Reviewer 2: ```suggestion
| [@ng-rsbuild/plugin-nx](https://github.com/Coly010/ng-rspack-build/tree/main/packages/build-plugin-nx)           | [@nx/angular](https://github.com/nrwl/nx/tree/master/packages/angular)                            | [@nx/angular](https://github.com/nrwl/nx/tree/master/packages/angular)                                  | Provides Nx integration for RsPack and RsBuild in Angular, like `@nx/angular` for Angular CLI. |
```

| **Configuration File (Bundler)**              | ✅    | [Rspack Config (`rspack.config.js`)](https://rspack.dev/config/)                                                | ✅    | [Angular JSON (`angular.json`)](https://angular.dev/guide/workspace-config)   | ✅    | [Webpack Config (`webpack.config.js`)](https://webpack.js.org/configuration/)                             | Angular combines bundler and builder configs in one file (`angular.json`)                                                   |
| **Configuration File (Builder)**              | ✅    | [Rsbuild Config (`rsbuild.config.ts`)](https://github.com/web-infra-dev/rsbuild)                                | ✅    | [Angular JSON (`angular.json`)](https://angular.dev/guide/workspace-config)   | ✅    | [Webpack Config (`webpack.config.js`)](https://webpack.js.org/configuration/)                             |                                                                                                                             |

> | **Compiler**                                  | ✅    | [@ng-rsbuild/compiler](https://github.com/Coly010/ng-rspack-build/tree/main/packages/compiler)                  | ✅    | [@angular/build](https://github.com/angular/angular/tree/main/packages/build) | ✅    | [@angular/compiler](https://github.com/angular/angular/tree/main/packages/compiler)                       |                                                                                                                             |

Reviewer 2: ```suggestion
| **Compiler**                                  | ✅    | [@ng-rspack/compiler](https://github.com/Coly010/ng-rspack-build/tree/main/packages/compiler)                  | ✅    | [@angular/build](https://github.com/angular/angular/tree/main/packages/build) | ✅    | [@angular/compiler](https://github.com/angular/angular/tree/main/packages/compiler)                       |                                                                                                                             |
```

| **Automatic Downleveling via `browserslist`** | ❌    | -                                                                                                               | ✅    | -                                                                             | ✅    | -                                                                                                         | Can be supported via [Rspack Target](https://rspack.dev/config/target#browserslist)                                         |
| **Build Flags (`NG_BUILD_MANGLE=0`, etc.)**   | ❌    | -                                                                                                               | ✅    | -                                                                             | ✅    | -                                                                                                         | No method to override SWC options                                                                                           |

> | **HMR (Hot Module Replacement)**              | ✅️   | [@ng-rsbuild/build](https://github.com/Coly010/ng-rspack-build/tree/main/packages/build)                        | ✅    | [@angular/build](https://github.com/angular/angular/tree/main/packages/build) | ✅    | [@angular-devkit/build-angular](https://github.com/angular/angular-cli/tree/main/packages/angular_devkit) |                                                                                                                             |

Reviewer 2: ```suggestion
| **HMR (Hot Module Replacement)**              | ⚠️   | [@ng-rspack/build](https://github.com/Coly010/ng-rspack-build/tree/main/packages/build)         [@ng-rsbuild/plugin-angular](https://github.com/Coly010/ng-rspack-build/tree/main/packages/rsbuild-plugin-angular)                | ✅    | [@angular/build](https://github.com/angular/angular/tree/main/packages/build) | ✅    | [@angular-devkit/build-angular](https://github.com/angular/angular-cli/tree/main/packages/angular_devkit) |                                                                                                                             |
```

| - Serve Application                           | ✅    | -                                                                                                               | ✅    | `ng serve`                                                                    | ✅    | `ng serve`                                                                                                |                                                                                                                             |
| - Build Application                           | ✅    | -                                                                                                               | ✅    | `ng build`                                                                    | ✅    | `ng build`                                                                                                |                                                                                                                             |

> | **Angular Plugin**                            | ✅    | [@ng-rspack/plugin-angular](https://github.com/Coly010/ng-rspack-build/tree/main/packages/build-plugin-angular) | ✅    | [@angular/build](https://github.com/angular/angular/tree/main/packages/build) | ✅    | [@angular-devkit/build-angular](https://github.com/angular/angular-cli/tree/main/packages/angular_devkit) |                                                                                                                             |

Reviewer 2: I don’t think this row makes sense or is needed

Author: Don't we have a serve and dev feature?

Reviewer 2: Oh the suggestion included too much.

The intended line is the Angular Plugin line

Author: removed

> | **Nx Plugin**                                 | ✅    | [@ng-rspack/plugin-nx](https://github.com/Coly010/ng-rspack-build/tree/main/packages/build-plugin-nx)           | 🔘    | [@nrwl/angular](https://github.com/nrwl/nx/tree/master/packages/angular)      | 🔘    | [@nrwl/angular](https://github.com/nrwl/nx/tree/master/packages/angular)                                  | Nx is a 3rd party                                                                                                           |

Reviewer 2: ```suggestion
| **Nx Plugin**                                 | ✅    | [@ng-rspack/plugin-nx](https://github.com/Coly010/ng-rspack-build/tree/main/packages/build-plugin-nx)           | 🔘    | [@nx/angular](https://github.com/nrwl/nx/tree/master/packages/angular)      | 🔘    | [@nx/angular](https://github.com/nrwl/nx/tree/master/packages/angular)                                  | Nx is a 3rd party                                                                                                           |
```

> | **Nx Integration**                            | ✅    | [@ng-rspack/nx](https://github.com/Coly010/ng-rspack-build/tree/main/packages/nx-plugin)                        | 🔘    | n/a                                                                           | 🔘    | n/a                                                                                                       | Already 3rd party ([`@nrwl/angular`](https://github.com/nrwl/nx/tree/master/packages/angular))                              |

Reviewer 2: ```suggestion
| **Nx Integration**                            | ✅    | [@ng-rspack/nx](https://github.com/Coly010/ng-rspack-build/tree/main/packages/nx-plugin)                        | 🔘    | n/a                                                                           | 🔘    | n/a                                                                                                       | Already 3rd party ([`@nx/angular`](https://github.com/nrwl/nx/tree/master/packages/angular))                              |
```

This table maps the key ng-rspack and ng-rsbuild packages to their equivalent Angular CLI components to show how Rspack and Rsbuild replace or mirror Angular CLI's Webpack-based system.

| ng-rspack / ng-rsbuild Package                                                                    | Equivalent in Angular CLI / DevKit                                                                                      | Description                                                                                        |
| ------------------------------------------------------------------------------------------------- | ----------------------------------------------------------------------------------------------------------------------- | -------------------------------------------------------------------------------------------------- |
| [@ng-rspack/build](https://github.com/Coly010/ng-rspack-build/tree/main/packages/build)           | [@angular-devkit/build-angular](https://github.com/angular/angular-cli/tree/main/packages/angular_devkit)               | Core build system for ng-rspack, similar to Angular CLI's Webpack-based builder.                   |

> | [@ng-rspack/compiler](https://github.com/Coly010/ng-rspack-build/tree/main/packages/build)        | [@angular/compiler]()                                                                                                   | Compiler for Angular applications using Rspack, like Angular's [@angular/compiler]().              |

Reviewer 2: ```suggestion
| [@ng-rspack/compiler](https://github.com/Coly010/ng-rspack-build/tree/main/packages/build)        | [@angular/compiler]()                                                                                                   | Compiler for Angular applications using Rspack, utilizing Angular's [@angular/compiler]() and [@angular/build]().              |
```
